opm/material/common/MathToolbox.hpp

```diff
@@ -230,10 +230,20 @@ template <class Evaluation, class Scalar>
 Evaluation constant(const Scalar& value)
 { return Opm::MathToolbox<Evaluation>::createConstant(value); }
 
+template <class Evaluation, int numVars, class Scalar>
```

Comment:
`numVars` must become a runtime parameter here, or else the whole excercise would be pointless. also, this function should only be available for Evaluations where the number of derivatives is runtime specified and conversely the functions for compile time Evaluations should only be available for these, i.e., please add `std::enable_if`.